### PR TITLE
Fix primary key unique constraint error extraction for MySQL

### LINKED ISSUE #15033: New item with manually entered string primary key does not show "Field has to be unique" error message
### Describe the Bug

When I create a new collection with the primary key field and select "manually entered string", it does not show a duplication message when I enter the same number - it does not allow only to save. I can not select the uniqueness field because a database error pops up.

### To Reproduce

https://user-images.githubusercontent.com/90528895/184174320-b57d7e11-4eec-4201-ae64-a8cfdd257b43.mp4

### Errors Shown

_No response_

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

16.14.1

### What database are you using?

MySQL

### What browser are you using?

Edge

### How are you deploying Directus?

locally

### PR DESCRIPTION
**Pull request by hanneskuettner**
## Scope

MySQL does some weird stuff for primary key unique constraints errors, where it doesn't actually tell us the primary key, but instead yells `PRIMARY` at us. Handle this case in the error extraction logic and propagate that info up the stack to where we actually have the info about the primary key field name. 

Additionally MySQL 5.7 won't even tell us the collection we're dealing with, so we have to add that in as well...

What's changed:

- Added `primaryKey` flag to the `RecordNotUniqueError` extensions to propagate error info
- Fixed handling for unique primary key constraints

## Potential Risks / Drawbacks

None

## Review Notes / Questions

- Should this sit this far up the stack, or should we pass more context to the `translateDatabaseError` function? I'm not sure 🤔 

---

Fixes #15033

### CODE REVIEW

File: api/src/services/items.ts [hunk 1]
```diff
@@ -217,7 +217,22 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
...
-				throw await translateDatabaseError(err);
+				const dbError = await translateDatabaseError(err);
+
+				if (isDirectusError(dbError, ErrorCode.RecordNotUnique) && dbError.extensions.primaryKey) {
+					// This is a MySQL specific thing we need to handle here, since MySQL does not return the field name
+					// if the unique constraint is the primary key
+					dbError.extensions.field = pkField?.field ?? null;
+
+					if (dbError.extensions.collection === null) {
+						// This is a special case for MySQL 5.7, where the collection is not returned for the primary key constraint
```
Comment (paescuj) [inline api/src/services/items.ts:228]: We do not support MySQL 5.7 any more, I think this can be dropped.

**Review by 2chiefk — APPROVED** [review]

File: api/src/services/items.ts (hunk 1 above)
Comment (rijkvanzanten) [reply api/src/services/items.ts:228]: You're correct. MySQL 5 is end of life completely https://endoflife.date/mysql

**Review by rijkvanzanten — APPROVED** [review]

**Comment by hanneskuettner** [comment]
Just as a note here, it might also be a MariaDB special case that is needed. I'll verify that on Monday and open a follow up PR if that's the case

**Comment by paescuj** [comment]
> Just as a note here, it might also be a MariaDB special case that is needed, since MariaDB apparently follow the 5.7 error convention (thouGH I can't remember exactly in this case). I'll verify that on Monday and open a follow up PR if that's the case

Do you mean the other MySQL/MariaDB related changes by any chance? Those are still preseved 👍

**Comment by hanneskuettner** [comment]
I saw that, but I vaguely remember (and judging by the error message processing) that MariaDB also doesn't return the collection the primary key constrained failed, which is why the special case was there. But I'll just have to verify that on Monday :)

**Pull request merged**